mod-di-converter-storage-server/src/main/resources/templates/db_scripts/associations-migration-gen-table/associations_migration.sql

```diff
@@ -0,0 +1,233 @@
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp" WITH SCHEMA public;
+
+/*
+This script will migrate old profile_associations to new general association table. The order of DML is important to ensure consistent
+state before and after migration.
+*/
+
+DO
+-- action_to_mapping_profiles: migration
+$$
+  DECLARE
+    r                  record;
+    action_wrapper_id  UUID;
+    mapping_wrapper_id UUID;
+  BEGIN
+    FOR r IN
+      select atm.id, atm.jsonb
+      from action_to_mapping_profiles atm
+      LOOP
+        -- get existing wrapper for action profile
+        select id
+        into action_wrapper_id
+        from profile_wrappers
+        where action_profile_id = (r.jsonb ->> 'masterProfileId')::uuid;
+
+        -- get existing wrapper for mapping profile
+        select id
+        into mapping_wrapper_id
+        from profile_wrappers
+        where mapping_profile_id = (r.jsonb ->> 'detailProfileId')::uuid;
+
+        -- insert into new association table
+        INSERT INTO profile_associations (id, job_profile_id, master_wrapper_id,
+            detail_wrapper_id, master_profile_id, detail_profile_id,
+            master_profile_type, detail_profile_type, detail_order, react_to) values
+            (r.id,
+            null,
+            action_wrapper_id,
+            mapping_wrapper_id,
+            (r.jsonb ->> 'masterProfileId')::uuid,
+            (r.jsonb ->> 'detailProfileId')::uuid,
+           'ACTION_PROFILE',
+           'MAPPING_PROFILE',
+            (r.jsonb ->> 'order')::int,
+            null
+           ) ON CONFLICT DO NOTHING;
+
+      END LOOP;
+      RAISE NOTICE 'PROFILES_MIGRATION:: migrated from action_to_mapping_profiles';
+  END
+$$;
+
+DO
+-- job_to_match_profiles: migration
+$$
+  DECLARE
+    r                record;
+    match_wrapper_id UUID;
+    job_wrapper_id   UUID;
+  BEGIN
+    FOR r IN
+      select jtm.id, jtm.jsonb
+      from job_to_match_profiles jtm
+      LOOP
+
+        -- get existing wrapper for job profile
+        select id
+        into job_wrapper_id
+        from profile_wrappers
+        where job_profile_id = (r.jsonb ->> 'masterProfileId')::uuid;
+
+        -- insert into new association table
+        INSERT INTO profile_associations (id, job_profile_id, master_wrapper_id,
+            detail_wrapper_id, master_profile_id, detail_profile_id,
+            master_profile_type, detail_profile_type, detail_order, react_to) values
+            (r.id,
+            null,
+            job_wrapper_id,
+            match_wrapper_id,
```

Comment:
the match_wrapper_id variable is not initialized, so it will cause the creation of association that does not refer to wrapper of match profile. We can insert matchWrapperId directly from the association from the old "job_to_match_profiles" table.

Reply:
forgot to push that change thank you! regarding to just iterating over old-associations and inserting its wrappers, i guess it will be a bit slower in comparison with getting wrappers directly from profile_wrappers because on profile_wrappers we have indexes, while in old-associations table not

Reply:
thanks for the suggestion Ruslan! made the corresponding changes

Reply:
thank you Javokhir!